apps/src/StudioApp.js

```diff
@@ -977,6 +977,44 @@ StudioApp.prototype.onReportComplete = function (response) {
   this.authoredHintsController_.finishHints(response);
 };
 
+/**
+ * @param {string} puzzleTitle
+ * @param {object} level
+ * @param {boolean} showHints
+ * @returns {React.element}
+ */
+StudioApp.prototype.getInstructionsContent_ = function (puzzleTitle, level, showHints) {
+  var isMarkdownMode = window.marked && level.markdownInstructions && this.LOCALE === ENGLISH_LOCALE;
+
+  var renderedMarkdown;
+
+  if (isMarkdownMode) {
+    var markdownWithImages = this.substituteInstructionImages(level.markdownInstructions);
+    renderedMarkdown = marked(markdownWithImages);
+  }
+
+  var authoredHints;
+  if (showHints) {
+    authoredHints = this.authoredHintsController_.getHintsDisplay();
+  }
+
+  return (
+    <Instructions
+      puzzleTitle={puzzleTitle}
```

Comment:
I'll update the function comment in `getInstructiosnContent_` to explain that puzzleTitle is optional. We could also add some custom validation to Instructions.jsx propTypes if we wanted that says we need to have one or the other.